src/EditorControls.js

```diff
@@ -119,7 +119,13 @@ class EditorControls extends Component {
           this.props.beforeAddTrace(payload);
         }
 
-        graphDiv.data.push(this.props.useAsDefaultTrace);
+        // can't use default prop because plotly.js mutates it:
+        // https://github.com/plotly/react-chart-editor/issues/509
+        graphDiv.data.push(
+          this.props.useAsDefaultTrace
```

Comment:
I think this is still vulnerable to this problem... I would go with a generator just to be sure.